src/api/api.ts

```diff
@@ -2,58 +2,96 @@ import { signInWithPopup, signOut } from 'firebase/auth';
 import {
   CollectionReference,
   DocumentData,
-  Firestore,
   QuerySnapshot,
+  deleteDoc,
   doc,
   getDoc,
   getDocs,
   query,
   setDoc,
   updateDoc,
   where,
+  limit,
 } from 'firebase/firestore';
 
+import { NamesDBCollection } from '../constants/enums';
+
 import { pathHome, pathSetNick } from 'constants/constants';
 import { auth, db, usersCollection } from 'firebase/firebase';
 import { addCurrentEmailId, setUser } from 'store/slices/mainSlice';
 import { AppDispatch } from 'store/store';
-import { ILogInUserArg, IUpdateUserArg, IUser } from 'types/types';
+import { ILogInUserArg, IUniversalObjectArguments, IUser } from 'types/types';
 
-export const addUser = async (value: IUser): Promise<void> => {
-  await setDoc(doc(usersCollection, value.id), value);
+export const setFirestoreData = async <T extends {}>(
+  collection: CollectionReference,
+  { id, values }: IUniversalObjectArguments<T>,
+): Promise<void> => {
+  if (!id) {
+    return;
+  }
+
+  await setDoc(doc(collection, id), { ...values });
 };
 
-export const getUsers = async (): Promise<QuerySnapshot<DocumentData, DocumentData>> => {
-  const data = await getDocs(usersCollection);
+export const deleteFirestoreDataById = async (
+  nameCollection: string,
+  id: string,
+): Promise<void> => {
+  await deleteDoc(doc(db, nameCollection, id));
+};
 
-  return data;
+export const getFirestoreDataById = async (
+  nameCollection: string,
+  id: string,
+): Promise<DocumentData | void> => {
+  const docRef = doc(db, nameCollection, id);
+  const docInfo = await getDoc(docRef);
+
+  return docInfo.data();
 };
 
-export const LogOut = async (): Promise<void> => {
-  await signOut(auth);
+export const updateFirestoreDataById = async <T extends {}>(
+  { id, values }: IUniversalObjectArguments<T>,
+  nameCollection: string,
+): Promise<void> => {
+  if (!id) {
+    return;
+  }
+  const userDoc = doc(db, nameCollection, id);
+
+  await updateDoc(userDoc, values);
+};
+
+export const getFirestoreData = async (
+  collection: CollectionReference,
+  limitNumber?: number,
+): Promise<QuerySnapshot> => {
+  let data;
+
+  if (limitNumber !== undefined && limitNumber > 0) {
+    data = await getDocs(query(collection, limit(limitNumber)));
+
+    return data;
+  }
+
+  data = await getDocs(collection);
+
+  return data;
 };
 
 export const checkFieldValueExists = async (
-  collectionPath: CollectionReference<DocumentData, DocumentData>,
+  collectionPath: CollectionReference,
   fieldName: string,
   value: string,
 ): Promise<boolean> => {
-  const collectionRef = collectionPath;
```

Comment:
https://prnt.sc/N1elA6T242cq  This function should be more versatile, we should be able to work with different collections. DB we should not pass as props, we have 1 bd and it should be imported directly in the api itself. Leave it as it is, since my new task is related to this issue, I will correct in this funky. But check the others.